schema/src/main/scala/org/constellation/schema/snapshot/SnapshotProposal.scala

```diff
@@ -0,0 +1,21 @@
+package org.constellation.schema.snapshot
+
+import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}
+import io.circe.{Decoder, Encoder}
+import org.constellation.schema.Id
+import org.constellation.schema.signature.Signable
+
+import scala.collection.SortedMap
+
+case class SnapshotProposal(override val hash: String, height: Long, reputation: SortedMap[Id, Double]) extends Signable
```

Comment:
How about adding here `peer: Id`? Of course we have `message.origin` in Gossip but peers also pull proposals  outside of Gossip from nodes. Without knowing the `Id` you can't verify the proposal

Reply:
:+1: 

A proposal must be identified with the peer 1:1.

Reply:
`Id` is a part of `HashSignature` which in turn is part of `Signed` case class, which is pushed through the gossip. So we should be able to verify the peer.